### PR TITLE
New object: Enemy Blocker

### PR DESCRIPTION
**Pull request by Savsish**
It's like Inivisble Wall but the player can pass it.

### CODE REVIEW

**Comment by tobbi** [comment]
@Savsish Can you squash the commits together so that you only have one commit?

**Comment by Hume2** [comment]
Well, no need to inherit MovingSprite, MovingObject is enough. And also no need to store the width/height separately, when it's stored in the bbox.

File: src/object/enemy_blocker.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+//  SuperTux
+//  Copyright (C) 2007 Kovago Zoltan <DirtY.iCE.hu@gmail.com>
```
Comment (maxteufel) [inline src/object/enemy_blocker.cpp]: There's something wrong with this copyright (same for the header). Please correct that.

Additionally, I'd recommend dedicate such small changes to [Public Domain](https://en.wikipedia.org/wiki/Public_domain) or using a license such as [CC0](https://creativecommons.org/about/cc0/) if you didn't steal the changes from other files (if that's the case, mark the original authors and use GPL, please).

File: src/object/enemy_blocker.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#ifndef HEADER_SUPERTUX_OBJECT_ENEMY_BLOCKER_HPP
+#define HEADER_SUPERTUX_OBJECT_ENEMY_BLOCKER_HPP
+
+#include "object/moving_sprite.hpp"
+#include "supertux/physic.hpp"
+
+class Player;
+
+/** An object that only solid for badguys. */
+class EnemyBlocker : public MovingSprite
```
Comment (Hume2) [inline src/object/enemy_blocker.hpp]: No need to inherit MovingSprite, use MovingObject instead.

File: src/object/enemy_blocker.hpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+{
+public:
+  EnemyBlocker(const ReaderMapping& lisp);
+
+  HitResponse collision(GameObject& other, const CollisionHit& hit);
+
+private:
+  Physic physic;
+  float width, height;
```
Comment (Hume2) [inline src/object/enemy_blocker.hpp]: No need for this, because it's never used.

File: src/object/enemy_blocker.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  lisp.get("height", height);
+  bbox.set_size(width, height);
+}
+
+HitResponse
+EnemyBlocker::collision(GameObject& other, const CollisionHit& )
+{
+  BadGuy* badguy = dynamic_cast<BadGuy*> (&other);
+  
+  if (badguy == 0)
```
Comment (Hume2) [inline src/object/enemy_blocker.cpp]: Please, do this via the function collides(GameObject*)

File: src/object/enemy_blocker.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+#include "util/reader_mapping.hpp"
+
+EnemyBlocker::EnemyBlocker(const ReaderMapping& lisp) :
+  MovingSprite(lisp, "images/objects/invisible/invisible.sprite", LAYER_TILES, COLGROUP_STATIC),
+  physic(),
+  width(32),
+  height(32)
+{
+  lisp.get("width", width);
+  lisp.get("height", height);
```
Comment (Hume2) [inline src/object/enemy_blocker.cpp]: "width" and "height" should be local variables there, not class members.

File: src/object/enemy_blocker.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "object/enemy_blocker.hpp"
+
+#include "badguy/badguy.hpp"
+#include "supertux/object_factory.hpp"
+#include "util/reader_mapping.hpp"
+
+EnemyBlocker::EnemyBlocker(const ReaderMapping& lisp) :
+  MovingSprite(lisp, "images/objects/invisible/invisible.sprite", LAYER_TILES, COLGROUP_STATIC),
```
Comment (Hume2) [inline src/object/enemy_blocker.cpp]: Do you really need to draw an empty sprite?

File: src/object/enemy_blocker.hpp [hunk 7]
```diff
@@ -0,0 +1,40 @@
+//  SuperTux
+//  Copyright (C) 2007 Kovago Zoltan <DirtY.iCE.hu@gmail.com>
```
Comment (Hume2) [inline src/object/enemy_blocker.hpp]: When you're at it, correct this as well.

File: src/object/enemy_blocker.hpp [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+/** An object that only solid for badguys. */
+class EnemyBlocker : public MovingSprite
+{
+public:
+  EnemyBlocker(const ReaderMapping& lisp);
+
+  HitResponse collision(GameObject& other, const CollisionHit& hit);
+
+private:
+  Physic physic;
```
Comment (Hume2) [inline src/object/enemy_blocker.hpp]: This variable is never used.

File: src/object/enemy_blocker.hpp [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HEADER_SUPERTUX_OBJECT_ENEMY_BLOCKER_HPP
+#define HEADER_SUPERTUX_OBJECT_ENEMY_BLOCKER_HPP
+
+#include "object/moving_sprite.hpp"
+#include "supertux/physic.hpp"
+
+class Player;
```
Comment (Hume2) [inline src/object/enemy_blocker.hpp]: This is useless.

File: src/object/enemy_blocker.hpp [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HEADER_SUPERTUX_OBJECT_ENEMY_BLOCKER_HPP
+#define HEADER_SUPERTUX_OBJECT_ENEMY_BLOCKER_HPP
+
+#include "object/moving_sprite.hpp"
+#include "supertux/physic.hpp"
```
Comment (Hume2) [inline src/object/enemy_blocker.hpp]: After you delete the variable "Physic", you don't need this include.

**Comment by tobbi** [comment]
@Savsish The idea is not bad. Can you please restore your branch and reopen this PR or open another PR? We just told you some code review steps.

**Comment by Savsish** [comment]
No longer vaild. I will create new pull request.

**Pull request closed without merging**